File: exampleretrofit/src/main/java/foo/bar/example/asafretrofit/api/fruits/FruitService.java
package foo.bar.example.asafretrofit.api.fruits;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

/**
 * These stubs are hosted at https://www.mocky.io/
 */
public interface FruitService {

    @GET("59efa0132e0000ef331c5f9b/")
    Call<List<FruitPojo>> getFruitsSimulateOk(@Query("mocky-delay") String delayScalaDurationFormat);

    @GET("59ef2a6c2e00002a1a1c5dea/")
    Call<List<FruitPojo>> getFruitsSimulateNotAuthorised(@Query("mocky-delay") String delayScalaDurationFormat);

}

File: example04retrofitcoroutine/src/main/java/foo/bar/example/foreretrofitcoroutine/api/fruits/FruitService.kt
package foo.bar.example.foreretrofitcoroutine.api.fruits

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

/**
 * These stubs are hosted at https://www.mocky.io/
 *
 * http://www.mocky.io/v2/59efa0132e0000ef331c5f9b
 * http://www.mocky.io/v2/59ef2a6c2e00002a1a1c5dea
 *
 */
interface FruitService {

    @GET("59efa0132e0000ef331c5f9b/")
    fun getFruitsSimulateOk(@Query("mocky-delay") delayScalaDurationFormat: String): Call<List<FruitPojo>>

    @GET("59ef2a6c2e00002a1a1c5dea/")
    fun getFruitsSimulateNotAuthorised(@Query("mocky-delay") delayScalaDurationFormat: String): Call<List<FruitPojo>>

}
